wolfssl/wolfcrypt/types.h

```diff
@@ -844,15 +844,15 @@ typedef struct w64wrapper {
         #if defined(HAVE_ECC) || defined(HAVE_OCSP) || \
         defined(WOLFSSL_KEY_GEN) || !defined(NO_DSA) || \
         defined(OPENSSL_EXTRA)
-            #define XTOUPPER(c)     toupper((c))
+            #define XTOUPPER(c)     toupper((unsigned char)(c))
```

Comment:
This won't work for all stdlib's. Some use `int` here. You might need to add a build option for the "type" used. Suggestion:

```
#ifndef STDC_CHAR_TYPE
#define STDC_CHAR_TYPE int
#endif
...

#define XTOUPPER(c) toupper((STDC_CHAR_TYPE)(c))
```

Reply:
From the looks of it (ctype implementation man page etc) I think the right approach here is to leave the macro wrappers unchanged (dangerous overkill to change globally), and in `ssl.c` `wolfSSL_ASN1_STRING_canon()` where octets are being used as arguments to `XISSPACE()`, use
```
XISSPACE((int)(unsigned char)(*dst))
```
etc.

Reply:
I discussed this with douzzer.  The original issue was only on XISSPACE, and we have since added casts in all calls of XISSPACE, so I don't believe universal casts are needed, and we believe they will lead to further issues.
Let me know if you disagree and think I should restore this with STDC_CHAR_TYPE.